### DISCUSSION TITLE
Solis Inverter Report locally

### DISCUSSION THREAD

**Discussion by Phil2204** [alienatedsec/solis-ha-modbus-cloud#16] [Q&A]
Hi all,

I am wondering if Someone can help me with my issue to get Data locally in my Network to use it for evcc. My inverter is connected via rs485 to the eastron smartmeter (w/o Modbus) and i have connected the w3 wifi Stick. Where is the Best way to get the data? Thanks for help.
Best regards Philipp

**Comment by alienatedsec** [comment]
@Phil2204 You could use my design and add another TCP client to connect additional equipment. 

Is there any documentation?

**Reply by Phil2204** [reply]
Thanks, this configureation Are u speaking about? https://github.com/alienatedsec/solis-ha-modbus-cloud/commits/v1.0.0 #

Which documentation?

**Reply by alienatedsec** [reply]
@Phil2204 i am quite familiar with the content I created. 

I meant an evcc and your requirements. I could analyse and advice further as I don't own any of it.

**Reply by Phil2204** [reply]
Evcc.io is  Software to work especially with Data from the inverter to charge to e car with the sun. Need Information from the inverter to use the Software on my rasp...

**Reply by Phil2204** [reply]
My inverter is solis rhi3p10k with BYD Battery . Energymeter is from eastron (sdm630MCT)

**Reply by alienatedsec** [reply]
> Evcc.io is Software to work especially with Data from the inverter to charge to e car with the sun

@Phil2204 In what format do you need this data that comes from the inverter?

**Reply by alienatedsec** [reply]
It looks like you need a template first - Solis inverters are using the same connectivity as Solax if combined with Waveshare ([this design](https://github.com/alienatedsec/solis-ha-modbus-cloud)). You could create your own based on this https://github.com/evcc-io/evcc/blob/master/templates/docs/meter/solax_0.yaml

I don't use the evcc software, but it looks easy to integrate.

**Reply by Phil2204** [reply]
@alienatedsec:  I had some time to went through your description and would like to implement this as well. Two questions: 

1) S3- Wifi Logging Stick is not working simultaneously with the local connection ( I am using currently S3)
2) Hybrid inverter (see above) is working with your setup as well?

thanks for supporting me.

**Reply by alienatedsec** [reply]
> S3- Wifi Logging Stick is not working simultaneously with the local connection ( I am using currently S3)

It will not work on its own as it is a cloud-only datalogger, so you could implement my design and use both, your S3 datalogger which reports to the Solis Cloud and Waveshare to send data locally to Home Assistant. 

> Hybrid inverter (see above) is working with your setup as well?

Any Solis inverter with a 4-pin connection - the same connection used by your datalogger.

**Reply by Phil2204** [reply]
I will buy all that required staff. what cable did you use? Only RJ45 Cable and/or twisted cable?  how did you fix the cable to the components? which connectors?

**Reply by alienatedsec** [reply]
@Phil2204 

I used an Ethernet cable (Cat 6) for RS485 - being pedantic here, the RJ45 is the type of connector where you could use an Ethernet cable too.

About connectors, please follow the guideline on the main page, especially the wiring diagram and the list of required components.

**Wiring**
https://github.com/alienatedsec/solis-ha-modbus-cloud#wiring

**All components**
https://github.com/alienatedsec/solis-ha-modbus-cloud#devices-and-cost

**Just connectors - Exceedconn EC04681-2023-BF Male/Female for Solis/Ginlong Inverter RS-485 port**
![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/eaffe218-5fcd-4784-95ab-7564731d263c)

**Reply by Phil2204** [reply]
Unterstand that you Are using twisted cable for Pin 3/4 . What kind of cable is needed for Pin1/2 ?

**Reply by alienatedsec** [reply]
> What kind of cable is needed for Pin1/2 ?

@Phil2204 you can use another pair from the same ethernet cable.

**Reply by Phil2204** [reply]
Ah, power for the waveshare is coming from the inverter?

**Reply by alienatedsec** [reply]
@Phil2204 I hope you can follow this diagram - if you don't have a PoE Switch, or even PoE capable Waveshare, you will need to use another source to power your Waveshare and that CANNOT BE the inverter.

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/d3e75597-0bfb-4488-9b65-3d7cec58b82f)

**Reply by Phil2204** [reply]
Hi @alienatedsec perfect, thanks for your help. I have almost all wired. the grounded cable is still missing. Is it ok to fix the grounded cable to the din rail within the sourcing map. where did you connect the grounded cable?

**Reply by alienatedsec** [reply]
@Phil2204 There is a difference between ground and earth, so please check carefully. The design highlights earthing but it's not needed.

You could attach it to a din rail or a junction connector. Up to you but follow the standard.

**Reply by Phil2204** [reply]
@alienatedsec: absolutely, I "only" need to ground the rs485? Why did you design the earthing?

**Reply by alienatedsec** [reply]
You don't need to and it's personal preference. There are benefits to each approach and I used earthing to protect the equipment and the environment it's connected into. 

“Only” Waveshare has an earth and a ground connector, so use them based on your preference.

**Reply by Phil2204** [reply]
@alienatedsec hi, i an wondering if you connected the Data logging Stick with the waveshare and everything works? I did that with your introduction but the connection to the inverter is Not working.  Best regards Philipp

**Reply by alienatedsec** [reply]
@Phil2204 please can you share config screenshots from VirCom of both waveshare devices?

**Reply by Phil2204** [reply]
Hi, sorry for delay. Now I am at home. Here my vircom Setup![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/127651372/9a35390c-dded-4a20-bdcb-89ca5d8cc42b)

Server:

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/127651372/c18129b8-0524-45df-8850-b782cd32079d)

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/127651372/eb481407-5cb8-4c57-8c17-e942e6e908cf)

Client:

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/127651372/86f11afb-c710-428b-973f-9a4e814ca892)

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/127651372/6434e524-2ff5-459d-97c2-fb6d6cf99445)

**Reply by alienatedsec** [reply]
I am wondering how clear the instructions are to follow and if I need to make those clearer. 

However, your TCP Client is wrongly configured. 

The below (highlighted in red) IP needs to be the IP of TCP Server which you covered on your screenshot and TCP port needs to be 502. 
![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/c22f3613-b6e0-445b-bdba-e314e0c293d6)

Another misconfiguration below. It should be a Simple Modbus setting. 

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/c51c30bd-3200-4885-a18a-9e4545489fc3)

**Comment by Phil2204** [comment]
Thanks @alienatedsec ! I will check that tomorrow. I really appreciate your Support...

**Reply by alienatedsec** [reply]
@Phil2204 I got a notification via email with your comment that something wasn't working - As I cannot see that comment anymore, did you get it resolved?

**Reply by Phil2204** [reply]
@alienatedsec : thanks for asking. I get still no connection to the data stick (yellow LED is only flashing sometimes, green and red light constantly). Even if I reset the data stick the connection to the inverter does not work.  I tried to check the baud rate setup for the inverter and did not find this information. Do you have an idea how to find that (inverter RHI 3P10K HVES) ? is there something else to check?

**Reply by alienatedsec** [reply]
@Phil2204 is anything working? E.g. do you get data from your tcp server to Home Assistant?

**Reply by alienatedsec** [reply]
@Phil2204 I recommend you revisit instructions and recheck your wiring based on my repo as you will have all answers there.

**Reply by Phil2204** [reply]
Ok , as you See I am a absolut begonnen with that stuff. I will check wiring etc later, thanks. I did not Start with Home assistant

**Reply by alienatedsec** [reply]
> I am a absolut begonnen with that stuff.

@Phil2204 your experience level doesn't matter and following the instructions will be the key. Make sure A and B connections are connected to the right pin and Wavehsare is configured as I described.
